Rename requireFeature to requireSuffix (#32051)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

Nested events are reflected in the HTML test reports, providing clear traceability.

#### Dependency declaration improvements

> 

Reviewer: ```suggestion

Gradle 8.11 introduced a new [requireFeature](https://docs.gradle.org/8.11/javadoc/org/gradle/api/artifacts/ModuleDependencyCapabilitiesHandler.html) API to select variants of a target component using the name of the feature.
```

> The `requireFeature` functionality added in Gradle 8.11 has been renamed to `requireSuffix`, 

Reviewer: ```suggestion
The `requireFeature` method has been renamed to `requireSuffix`, 
```

allowing any arbitrary suffix to be appended to the coordinates of a module as a capability.

> The `feature` terminology has been repurposed as a shorthand to a new dependency modifier, similar to `textFixtures`,

Reviewer: ```suggestion
The `feature` terminology has been repurposed as a shorthand for a new dependency modifier, similar to `testFixtures`.
```

> allowing a dependency on a feature of a dependency to be declared. 

Reviewer: ```suggestion
It enables declaring a dependency on a specific feature of a module.
```

> The `feature` dependency modifier automatically converts the camel case feature name (`featureName`) to the kebob-case suffix (`-feature-name`).

Reviewer: ```suggestion
The `feature` dependency modifier automatically converts a camelCase feature name (e.g., `featureName`) into a kebab-case suffix (e.g., -feature-name).
```

> Given some feature declared in a Java project:

Reviewer: ```suggestion
Suppose a feature is declared in a Java project as follows:

```

```groovy
plugins {
    id("java-library")
}

java {
    registerFeature("featureName") {
        usingSourceSet(sourceSets.create("featureName"))
    }
}
```

> The feature dependency modifier can be used to declare a dependency on the feature:

Reviewer: ```suggestion
You can use the feature modifier to declare a dependency on this feature:
```
